### PR TITLE
fix uploads of >2MB files to Pulp3-based setups

### LINKED ISSUE #1043: theforeman.foreman.content_upload fails with error "non_field_errors: Chunk size does not match content range."
##### SUMMARY

`theforeman.foreman.content_upload` fails with error `"non_field_errors: Chunk size does not match content range."` when uploading [this binary file](https://github.com/docker/compose/releases/download/1.27.4/docker-compose-Linux-x86_64)

The same tasks works ok when uploading a small txt file instead of a binary file

The file triggering the error can be uploaded using hammer doing `hammer repository upload-content --organization-id 1 --id 1548 --content-type file --path /tmp/docker-compose-Linux-x86_64`  . Hammer upload test is done from the same host where katello is running. Ansible task triggering the error is executed from a different host.

##### ISSUE TYPE
 - Bug Report

##### ANSIBLE VERSION

```
ansible 2.10.2
  config file = /home/escobar/ansible/playbooks/ansible-playbook-foreman-katello/ansible.cfg
  configured module search path = ['/home/escobar/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/escobar/tmp/venv_ansible_2.10.2/lib/python3.6/site-packages/ansible
  executable location = /home/escobar/tmp/venv_ansible_2.10.2/bin/ansible
  python version = 3.6.8 (default, Nov 21 2019, 19:31:34) [GCC 8.3.1 20190507 (Red Hat 8.3.1-4)]

```

##### COLLECTION VERSION
```
# /home/escobar/ansible/playbooks/ansible-playbook-foreman-katello/collections/ansible_collections
Collection         Version
------------------ -------
theforeman.foreman 1.4.0
```

##### KATELLO/FOREMAN VERSION
```
tfm-rubygem-katello-3.16.0-1.el7.noarch
foreman-2.1.2-1.el7.noarch
```

##### STEPS TO REPRODUCE

```
wget -P /tmp/ https://github.com/docker/compose/releases/download/1.27.4/docker-compose-Linux-x86_64
```

```yaml
- name: "Create product 'docker-compose'"
  theforeman.foreman.product:
    username: "{{ foreman_username }}"
    password: "{{ foreman_password }}"
    server_url: "{{ foreman_url }}"
    validate_certs: "{{ foreman_validate_certs }}"
    organization: "{{ foreman_initial_organization }}"
    name: "docker-compose"
    state: present

- name: "Create files repository 'docker-compose-linux-x86_64' for product 'docker-compose'"
  theforeman.foreman.repository:
    username: "{{ foreman_username }}"
    password: "{{ foreman_password }}"
    server_url: "{{ foreman_url }}"
    validate_certs: "{{ foreman_validate_certs }}"
    organization: "{{ foreman_initial_organization }}"
    name: "docker-compose-linux-x86_64"
    state: present
    content_type: "file"
    product: "docker-compose"
  register: _repo_info

- name: "Upload a test file"
  theforeman.foreman.content_upload:
    username: "{{ foreman_username }}"
    password: "{{ foreman_password }}"
    server_url: "{{ foreman_url }}"
    validate_certs: "{{ foreman_validate_certs }}"
    organization: "{{ foreman_initial_organization }}"
    src: "/tmp/docker-compose-Linux-x86_64"
    #src: "/tmp/lele.txt"
    product: "docker-compose"
    repository: "docker-compose-linux-x86_64"

```

##### EXPECTED RESULTS

File should be uploaded to the repository but it fails. The same task works OK when uploading a small txt file. e.g

```
echo "test" > /tmp/lelele.txt
```

```yaml
- name: "Upload a test file"
  theforeman.foreman.content_upload:
    username: "{{ foreman_username }}"
    password: "{{ foreman_password }}"
    server_url: "{{ foreman_url }}"
    validate_certs: "{{ foreman_validate_certs }}"
    organization: "{{ foreman_initial_organization }}"
    src: "/tmp/lelele.txt"
    product: "docker-compose"
    repository: "docker-compose-linux-x86_64"
```

### PR DESCRIPTION
**Pull request by evgeni**
Fixes: #1043

### CODE REVIEW

**Comment by evgeni** [comment]
Yes, I tested this against a Pulp2-based Katello 3.12 (don't ask) and it still works ;)

File: plugins/modules/content_upload.py [hunk 1]
```diff
@@ -174,7 +174,7 @@ def main():
 
                 with open(b_src, 'rb') as contentfile:
                     for chunk in iter(lambda: contentfile.read(CONTENT_CHUNK_SIZE), b""):
-                        data = {'content': chunk, 'offset': offset, 'size': len(chunk)}
+                        data = {'content': chunk, 'offset': offset, 'size': size}
```
Comment (mdellweg) [inline plugins/modules/content_upload.py:177]: Did you check, this does not blow up if you chunk the file into more that two pieces?

Comment (evgeni) [reply plugins/modules/content_upload.py:177]: Yes, I tested it with the file from #1043 which is 6 (well, 5.8) chunks big

Comment (mdellweg) [reply plugins/modules/content_upload.py:177]: Strange but ok.
I think the actual bug is here:
https://github.com/Katello/katello/blob/540e695c5e108df99d84acae620ab641b6c898f0/app/services/katello/pulp3/content.rb#L32
The size parameter for the content_range should be taken from the upload object, not the update_upload request.

Comment (mdellweg) [reply plugins/modules/content_upload.py:177]: that said again, pulp upload does not even use or expect the size field (it's marked read_only, but never actually set).

**Review by mdellweg — APPROVED** [review]

File: plugins/modules/content_upload.py (hunk 1 above)
Comment (evgeni) [reply plugins/modules/content_upload.py:177]: Yeah, I also wondered why the chunk upload needs the size at all, but oh well… I'll open a refactor issue against katello

Comment (evgeni) [reply plugins/modules/content_upload.py:177]: https://projects.theforeman.org/issues/31300

**Pull request merged**